src/OpenSSL/SSL.py

```diff
@@ -1914,6 +2045,10 @@ def connect(self, addr):
         :param addr: A remote address
         :return: What the socket's connect method returns
         """
+        if self.is_datagram_socket:
+            # DTLS BIO set connected
+            _lib.BIO_ctrl(_lib.SSL_get_rbio(self._ssl),
+                          _lib.BIO_CTRL_DGRAM_SET_CONNECTED, 0, addr[0])
```

Comment:
Drive-by comment since I was fiddling with the bindings side. This doesn't look right. See the discussion here:
https://github.com/pyca/cryptography/pull/4220#issuecomment-388480321